hooks/useSwapPositionsData/useSwapPositionsData.ts

```diff
@@ -0,0 +1,278 @@
+import { CurrencyAmount, Price, Token } from '@uniswap/sdk-core';
+import { priceToClosestTick, TickMath } from '@uniswap/v3-sdk';
+import { ethers } from 'ethers';
+import { getAddress } from 'ethers/lib/utils.js';
+import { ActivityType } from 'hooks/useActivity/useActivity';
+import { useConfig } from 'hooks/useConfig';
+import { useController } from 'hooks/useController';
+import { useLatestMarketPrice } from 'hooks/useLatestMarketPrice';
+import { price } from 'lib/controllers/charts/mark';
+import { computeDeltasFromActivities } from 'lib/controllers/uniswap';
+import { erc20TokenToToken, uniTokenToErc20Token } from 'lib/uniswapSubgraph';
+import { useMemo } from 'react';
+import {
+  LpActivityByAccountDocument,
+  LpActivityByAccountQuery,
+  SwapActivityByAccountDocument,
+  SwapActivityByAccountQuery,
+} from 'types/generated/graphql/inKindSubgraph';
+import { useQuery } from 'urql';
+
+export function useSwapPositionsData(
+  address: string | undefined,
+  startTimestamp: number,
+  endTimestamp: number,
+) {
+  const { chainId } = useConfig();
+  const { paprToken, token0IsUnderlying, underlying } = useController();
+  const price = useLatestMarketPrice();
+
+  const [{ data: swapActivityForUserData }] =
+    useQuery<SwapActivityByAccountQuery>({
+      query: SwapActivityByAccountDocument,
+      variables: {
+        account: address?.toLowerCase(),
+        startTimestamp: startTimestamp,
+        endTimestamp: endTimestamp,
+      },
+      pause: !address,
+    });
+
+  const [{ data: lpActivityForUserData }] = useQuery<LpActivityByAccountQuery>({
+    query: LpActivityByAccountDocument,
+    variables: {
+      account: address?.toLowerCase(),
+      startTimestamp: startTimestamp,
+      endTimestamp: endTimestamp,
+    },
+    pause: !address,
+  });
+
+  const { token0, token1 } = useMemo(() => {
+    if (token0IsUnderlying)
+      return {
+        token0: erc20TokenToToken(underlying, chainId),
+        token1: erc20TokenToToken(paprToken, chainId),
+      };
+    else
+      return {
+        token0: erc20TokenToToken(paprToken, chainId),
+        token1: erc20TokenToToken(underlying, chainId),
+      };
+  }, [token0IsUnderlying, underlying, paprToken, chainId]);
+
+  // returns users swap activities with pseudo swaps for liquidity positions
+  const swapsWithImplicit = useMemo(() => {
```

Comment:
this is big enough that it would benefit from factoring out of the main body of the hook into helper functions